Feature/issue 17

9noeyni9 opened the pull request:
### ✅ PR 종류

- [x] 기능 추가
- [x] 버그 수정
- [x] 리팩토링
- [ ] 테스트
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

--- 

### ⛏ 반영 브랜치

- Feature/issue 17 -> dev

---

### 📑 변경 사항

- 마이페이지 비밀번호 변경 구현
- 최근 3번 사용 기록이 있는 비밀번호 재사용 제한
- 인코딩 된 값으로 DB에 저장 되도록 수정
- PasswordHistoryRepository query method에 parameter 수정(user -> userId)
- controller에 return 형식 ResponseDto로 수정 (한 줄 소개 수정 부분, 비밀번호 수정 부분)

---

### 👀 중요하게 확인할 부분

- 가독성이 많이 떨어지는 거 같아 코드 설명 주석 추가했습니다.. 머지 전 삭제하도록 하겠습니다 

---

### ❗ 관련 이슈

close #17

9noeyni9 committed:
feat : 마이페이지 비밀번호 변경 구현

- User, UserService 수정
- 변경 사항 추가 예정

9noeyni9 committed:
fix : 마이페이지 비밀번호 수정 코드 수정 중

9noeyni9 committed:
refactor : 충돌 해결 merge

9noeyni9 committed:
refactor : return 형식 ResponseDto로 수정

- 한 줄 소개 수정 부분
- 비밀번호 수정 부분

9noeyni9 committed:
bug : 비밀번호 일치 오류 해결

- 인코딩 된 값으로 DB에 저장 되도록 수정
- PasswordHistoryRepository query method에 parameter 수정(user -> userId)
- 코드 설명 주석 추가 (머지 전 삭제 예정)

9noeyni9 committed:
refactor : 불필요한 import문 제거

jinkshower commented on src/main/java/com/spring/nbcijo/service/MyPageService.java:
```diff
@@ -40,18 +44,63 @@ public void updateMyDescription(User user,
+
+    public boolean isPasswordMatches(String passwordInDB, String inputPassword) {
+        return passwordEncoder.matches(inputPassword, passwordInDB);
+    }
```
이 클래스 안에서만 쓰이는것같은데 private으로 만드는 건 어떤가요?

jinkshower replied:
밑의 메서드도 마찬가지입니다

jinkshower commented on src/main/java/com/spring/nbcijo/service/MyPageService.java:
```diff
@@ -40,18 +44,63 @@ public void updateMyDescription(User user,
+                    ? 1 : 0;
+        }
+        result = count == 0 ? true : false;
+        return result;
```
return count == 0 으로 해도 true false가 넘어갈것같네요

9noeyni9 replied:
아이고 그렇네요 다른 리뷰랑 같이 수정해서 다시 commit 하겠습니다 감사합니다~

jinkshower commented on src/main/java/com/spring/nbcijo/service/MyPageService.java:
```diff
@@ -40,18 +44,63 @@ public void updateMyDescription(User user,
+        boolean result = true;
+        List<PasswordHistory> passwordList = passwordHistoryRepository.findTop3ByUserIdOrderByIdDesc(
+            user.getId());
+        System.out.println("service passwortList.size() : " + passwordList.size());
```
디버깅 메서드 인가요? 로그에 찍혀야 하는 정보면 log를 사용해주시면 좋을 것같아요

jinkshower commented on src/main/java/com/spring/nbcijo/service/MyPageService.java:
```diff
@@ -40,18 +44,63 @@ public void updateMyDescription(User user,
+        for (PasswordHistory password : passwordList) {
+            count +=
+                isPasswordMatches(password.getPassword(), updatePasswordRequestDto.getNewPassword())
+                    ? 1 : 0;
```
삼항 연산자 보다 그냥 if를 쓰는게 좀더 가독성이 좋을 것같은데 어떻게 생각하세요?

jinkshower commented on src/main/java/com/spring/nbcijo/service/UserService.java:
```diff
@@ -16,6 +19,9 @@ public class UserService {
     private final PasswordEncoder passwordEncoder;
+    private final PasswordHistoryRepository passwordHistoryRepository;
+    private PasswordHistory passwordHistory;
+    private User user;
```
다른 서비스는 모두 메서드안에서 객체를 생성하고 있는데요 , 인스턴스 변수가 이 클래스에만 있어서 일관성이 떨어지는것같아요.

9noeyni9 committed:
refactor : 리뷰 피드백 반영

- 한 클래스 내에서만 사용하는 메서드 private으로 변경
- 불필요한 출력문 제거
- isPasswordPreviouslyUsed 메서드 코드 변경
- UserService 메서드 내에서 객체 생성하도록 변경

9noeyni9 committed:
Merge branch 'dev' into feature/issue-17
